endpoints/openrtb2/video_auction.go

```diff
@@ -659,27 +654,32 @@ func (deps *endpointDeps) validateVideoRequest(req *openrtb_ext.BidRequestVideo)
 		}
 	}
 
-	if len(req.Video.Mimes) == 0 {
-		err := errors.New("request missing required field: Video.Mimes")
-		errL = append(errL, err)
-	} else {
-		mimes := make([]string, 0, 0)
-		for _, mime := range req.Video.Mimes {
-			if mime != "" {
-				mimes = append(mimes, mime)
+	if req.Video != nil {
+		if len(req.Video.MIMEs) == 0 {
+			err := errors.New("request missing required field: Video.Mimes")
+			errL = append(errL, err)
+		} else {
+			mimes := make([]string, 0, 0)
```

Comment:
I know this is copied code, but consider initializing to the size of `req.Video.MIMEs`.

Reply:
After playing around with this section, I think it may still be necessary to initialize with an empty slice. This check here filters out empty strings from `MIMEs`, so it is a different error than having no `MIMEs` in the request. Let me know if there is a better way to do this however.

Reply:
The slice would still be empty, but setting an initial capacity to `len(req.Video.MIMEs)` is more efficient if there is a high chance that all MIMEs are good. If you expect most of the MIME values to be empty, then leaving it this way is fine.

Reply:
Ah I just realized I set the initial length instead of the initial capacity when I originally tried to change this. You're right, working as expected. Updated so that initial cap is `len(req.Video.MIMEs)`.